### PR TITLE
[#251] Reorganize readme

### LINKED ISSUE #251: Simplify repository REAME
# Clarification and motivation

Currently the README of this repository is quite big and contains a lot of information.

However, most people are probably interested in only _a part_ of these info, moreover this makes the README quite overwhelming to look at.

We should make the README simpler and move specific details to separate `docs` to link at. 
_it's also probably a good idea to take this opportunity to clean it up an reorganize it a bit, if it makes sense_

# Acceptance criteria

- the README is shortened and simplified, linking to other documents instead
- several `docs` contain more detailed information, separate for each use-case

### PR DESCRIPTION
**Pull request by rvem**
## Description
Problem: At the moment main README.md is quite large and it's hard to
navigate through it, even though it has a table of contents.
Usually, users are interested in only one way of distribution and it's
not convenient for them to skim through the other distribution options.

Solution: Split README into multiple docs:
* README itself that points to the other docs.
* Ubuntu-related instructions.
* Fedora-related instructions.
* macOS-related instructions.
* Doc about systemd services.
* Doc about launchd services.

## Related issue(s)

Resolves #251

#### Related changes (conditional)

- [x] I checked whether I should update the [README](../../tree/master/README.md)

- [x] I checked whether native packaging works, i.e. native binary packages
  can be successfully built.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,259 +28,35 @@ different networks, you can read more about this in [this article](https://tezos
...
-Recomended way to get these binaries is to download them from assets from github release.
-Go to the [latest release](https://github.com/serokell/tezos-packaging/releases/latest)
-and download desired assets.
+The following distributions are supported by the `tezos-packaging`:
 
-Some of the individual binaries contain protocol name to determine
-with which protocol binary is compatible with. If this is not the
-case, then consult release notes to check which protocols are
-supported by that binary.
+* Ubuntu. Install packages using `apt-get` following [these instructions](./docs/distros/ubuntu.md).
```
Comment (ixahedron) [inline README.md:42]:
```suggestion
* **Ubuntu**. Install packages using `apt-get` following [these instructions](./docs/distros/ubuntu.md).
```

File: README.md [hunk 2]
```diff
@@ -28,259 +28,35 @@ different networks, you can read more about this in [this article](https://tezos
...
-./scripts/build-bottles.sh
-```
-
-Note that this might take a while, because builds don't share common parts and for each binary
-dependencies are compiled from scratch. Once the bottles are built, the corresponding sections in the
-formulas should be updated. Also, bottles should be uploaded to the release artifacts.
+  This option also supports Raspberry Pi OS, see [this part of the doc](./docs/distros/ubuntu.md#raspberry).
+* Fedora. Install packages using `dnf` or `yum` following [these instructions](./docs/distros/fedora.md).
+* Other linux distros can use static binaries from the [latest release](https://github.com/serokell/tezos-packaging/releases/latest).
+* macOS. Install formulaes using `brew` following [these instructions](./docs/distros/macos.md).
```
Comment (ixahedron) [inline README.md:47]:
```suggestion
  This option also supports **Raspberry Pi OS**, see [this part of the doc](./docs/distros/ubuntu.md#raspberry).
* **Fedora**. Install packages using `dnf` or `yum` following [these instructions](./docs/distros/fedora.md).
* Other linux distros can use static binaries from the [latest release](https://github.com/serokell/tezos-packaging/releases/latest).
* **macOS**. Install formulaes using `brew` following [these instructions](./docs/distros/macos.md).
```
I suggest to add bold emphasis, otherwise I found it hard to skim through.

**Review by ixahedron — APPROVED** [review]
Other than the minor formatting suggestion, LGTM.

File: README.md [hunk 3]
```diff
@@ -28,259 +28,35 @@ different networks, you can read more about this in [this article](https://tezos
...
-./scripts/build-bottles.sh
-```
-
-Note that this might take a while, because builds don't share common parts and for each binary
-dependencies are compiled from scratch. Once the bottles are built, the corresponding sections in the
-formulas should be updated. Also, bottles should be uploaded to the release artifacts.
+  This option also supports **Raspberry Pi OS**, see [this part of the doc](./docs/distros/ubuntu.md#raspberry).
+* **Fedora**. Install packages using `dnf` or `yum` following [these instructions](./docs/distros/fedora.md).
+* Other linux distros can use static binaries from the [latest release](https://github.com/serokell/tezos-packaging/releases/latest).
+* **macOS**. Install formulaes using `brew` following [these instructions](./docs/distros/macos.md).
```
Comment (pasqu4le) [inline README.md:47]: A mostly aesthetic suggestion I'd make is to make a heading for each entry, which is itself the link, and only put a small description below it.

Here my suggestion for this whole section would be:
```
<a name="getting-binaries"></a>
## Getting Tezos binaries

The following distributions are supported by the `tezos-packaging`:

### [macOS](./docs/distros/macos.md)

`brew` formulae and taps for stable releases and release candidates.

### [Ubuntu](./docs/distros/ubuntu.md)

Native Ubuntu packages that can be installed with `apt-get` from a PPA.

### [Rasperry Pi OS](./docs/distros/ubuntu.md#raspberry)

On some Raspberry systems Ubuntu packages can be installed using `apt-get`.

### [Fedora](./docs/distros/fedora.md)

Native Fedora packages that can be installed using using `dnf` or `yum`.

### [Static binaries and other linux distros](https://github.com/serokell/tezos-packaging/releases)

Prebuilt static binaries can be downloaded directly from the [latest release](https://github.com/serokell/tezos-packaging/releases/latest).
```

I've see this done in some other READMEs and I think it can be easier to read.

File: README.md [hunk 4]
```diff
@@ -28,259 +28,35 @@ different networks, you can read more about this in [this article](https://tezos
...
-* [Static linux binaries](#static-linux)
-* [Native Ubuntu packages](#ubuntu)
-  * [Ubuntu packages on Raspberry Pi OS](#raspberry)
-* [Native Fedora packages](#fedora)
-* [Other linux](#linux)
-* [Brew tap for macOS](#macos)
-* [Systemd services for Tezos binaries](#systemd)
-* [Building instructions](#building)
-* [Setting up baking instance on Ubuntu](#baking-on-ubuntu)
+* [Getting binaries](#getting-binaries).
```
Comment (pasqu4le) [inline README.md:31]: If you take my previous suggestion this could have sub-entries IMO.

File: README.md [hunk 5]
```diff
@@ -28,259 +28,35 @@ different networks, you can read more about this in [this article](https://tezos
...
+* [Systemd services with Tezos binaries](#systemd).
+* [Launchd services with Tezos binaries](#systemd).
+* [Setting up baking instance on Ubuntu](#baking-on-ubuntu).
+* [Building instructions](#building).
```
Comment (pasqu4le) [inline README.md:36]: Maybe contributing and about should have an entry here too?

File: README.md [hunk 6]
```diff
@@ -28,259 +28,35 @@ different networks, you can read more about this in [this article](https://tezos
...
-  * [Ubuntu packages on Raspberry Pi OS](#raspberry)
-* [Native Fedora packages](#fedora)
-* [Other linux](#linux)
-* [Brew tap for macOS](#macos)
-* [Systemd services for Tezos binaries](#systemd)
-* [Building instructions](#building)
-* [Setting up baking instance on Ubuntu](#baking-on-ubuntu)
+* [Getting binaries](#getting-binaries).
+* [Systemd services with Tezos binaries](#systemd).
+* [Launchd services with Tezos binaries](#systemd).
```
Comment (pasqu4le) [inline README.md:33]: I'm thinking that perhaps the `Launchd` document should be moved inside the `macOS` one.

From its content we only provide the former in the formulae for the latter, plus it makes sense to me that only `macOS` users might be interested in it.

For `sysyemd` packages given that they are provided for both native releases and can be build from scratch I'd leave those instructions in a separate doc (as they are now), but maybe instead of linking it from here we could add a link in the two distro docs as well as under the "other distros" section below?

File: docs/distros/macos.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+It's possible to provide prebuilt macOS packages for brew called bottles. They're supposed
+to be built before making the new release and included to it. In order to build all bottles run
+`build-bottles.sh` script:
+```
+./scripts/build-bottles.sh
+```
+
+Note that this might take a while, because builds don't share common parts and for each binary
+dependencies are compiled from scratch. Once the bottles are built, the corresponding sections in the
+formulas should be updated. Also, bottles should be uploaded to the release artifacts.
```
Comment (pasqu4le) [inline docs/distros/macos.md:38]: I'd reword this a little bit, this is mainly going to be seen by users, who aren't really interested in this part.

I'd change it to first mention that bottles are provided with every release for faster installation and then only mention that if one wants to build them there is a script to do so.

File: docs/distros/ubuntu.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
+<!--
+   - SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+   -
+   - SPDX-License-Identifier: LicenseRef-MIT-TQ
+   -->
+# Ubuntu Launchpad PPA with `tezos-*` binaries
+
+If you are using Ubuntu you can use PPA in order to install `tezos-*` executables.
```
Comment (pasqu4le) [inline docs/distros/ubuntu.md:8]: Given that it provides this and more info perhaps it makes sense to link the baking guide here?

File: docs/launchd.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+To start the service: `brew services start <formula>`.
+
+To stop the service: `brew services stop <formula>`.
+
+All of the brew services have various configurable env variables. These variables
+can be changed in the corresponding `/usr/local/Cellar/tezos-signer-tcp/<version>/homebrew.mxcl.<formula>.plist`.
+Once the configuration is updated, you should restart the service:
+`brew services restart <formula>`.
+
+Note, that all services are run as a user agents, thus they're stopped after the logout.
```
Comment (pasqu4le) [inline docs/launchd.md:33]: Perhaps this should be put in a bit more evidence, as it can be quite painful it if happens by surprise.

File: README.md [hunk 10]
```diff
@@ -28,259 +28,35 @@ different networks, you can read more about this in [this article](https://tezos
...
+* [Setting up baking instance on Ubuntu](#baking-on-ubuntu).
```
Comment (pasqu4le) [inline README.md:34]: Maybe change it to:
```suggestion
* [Setting up a node and/or baking on Ubuntu](#baking-on-ubuntu).
```

File: docs/distros/ubuntu.md (hunk 8 above)
Comment (pasqu4le) [reply docs/distros/ubuntu.md:8]: Additional note: the section about "`tezos-baking` package" should take the suggestion above and I think also be moved on the top of the doc.

File: README.md (hunk 10 above)
Comment (pasqu4le) [reply README.md]: You updated the link but not the header's content.

File: README.md [hunk 11]
```diff
@@ -28,259 +28,49 @@ different networks, you can read more about this in [this article](https://tezos
...
-# or use yum
-sudo yum copr enable @Serokell/Tezos
-sudo yum install tezos-baker-010-PtGRANAD
-```
-Once you install such packages the commands `tezos-*` will be available.
+Native Fedora packages that can be installed using using `dnf` or `yum`.
 
 <a name="linux"></a>
-## Other Linux distros usage
+### [Static binaries and other linux distros]
```
Comment (pasqu4le) [inline README.md:63]: Uh, this without a link _and_ between square brackets is odd, why not link this to the `/releases` page?

File: docs/distros/macos.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+If you're using macOS and `brew`, you can install Tezos binaries from the taps provided
+by this repository. There are two taps: one for the latest stable release of Tezos and
+one for the latest release candidate of Tezos.
+
+In order to use latest stable version run the following:
+```
+brew tap serokell/tezos-packaging-stable https://github.com/serokell/tezos-packaging-stable.git
+```
+
+In order to use latest release candidate version run the following:
```
Comment (pasqu4le) [inline docs/distros/macos.md:17]:
```suggestion
In order to use the latest release candidate version run the following:
```

File: docs/distros/macos.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+```
+brew tap serokell/tezos-packaging-rc https://github.com/serokell/tezos-packaging-rc.git
+```
+
+Once the desired tap is selected, you can install the chosen package, e.g.:
+```
+brew install tezos-client
+```
+
+For faster formulaes installation we provide prebuilt bottles for some macOS versions in the releases.
```
Comment (pasqu4le) [inline docs/distros/macos.md:27]:
```suggestion
For faster formulae installation we provide prebuilt bottles for some macOS versions in the releases.
```

File: docs/distros/macos.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+## Launchd background services on macOS.
+
+All provided `launchd` services are run as a user agents, thus they're stopped after the logout.
```
Comment (pasqu4le) [inline docs/distros/macos.md:31]:
```suggestion
IMPORTANT: All provided `launchd` services are run as a user agents, thus they're stopped after the logout.
```

File: docs/distros/macos.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+`tezos-accuser-<proto>`, `tezos-baker-<proto>`, `tezos-endorser-<proto>` formulas
+provide backround services for running the corresponding daemons.
+
+Since `tezos-node` and `tezos-signer` need multiple services they are provided
+in dedicated meta-formulas. These formulas don't install any binaries and only add
+background services.
+
+Formulas with `tezos-node` background services:
+* `tezos-node-mainnet`
```
Comment (pasqu4le) [inline docs/distros/macos.md:41]: This list seems a bit short.

File: docs/systemd.md [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+<a name="ubuntu-and-fedora"></a>
+## Systemd units on Ubuntu and Fedora
+
+`tezos-node`, `tezos-accuser-<proto>`, `tezos-baker-<proto>`,
+`tezos-endorser-<proto>`, and `tezos-signer` packages have systemd files included to the
+Ubuntu and Fedora packages.
+
+Once you've installed the packages with systemd unit, you can run the service
+with the binary from the package using the following command:
```
Comment (pasqu4le) [inline docs/systemd.md:16]:
```suggestion
Once you've installed one of these packages, you can run the service
using the following command:
```
Seems a bit redundant otherwise

File: docs/systemd.md [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+```
+systemctl start <package-name>.service
+```
+To stop the service run:
+```
+systemctl stop <package-name>.service
+```
+
+Each service has configuration file located in `/etc/default`. Default
```
Comment (pasqu4le) [inline docs/systemd.md:25]:
```suggestion
Each service has its configuration file located in `/etc/default`. Default
```

File: docs/systemd.md [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+<a name="generic-linux"></a>
+## Systemd units on other Linux systems
+
+If you're not using Ubuntu or Fedora you can still construct systemd units for binaries
+from scratch.
+
+For this you'll need `.service` file to define systemd service. The easiest way
+to get one is to run [`gen_systemd_service_file.py`](../gen_systemd_service_file.py).
+You should specify service name as an argument. Note that there are three
+predefined services for `tezos-node`: `tezos-node-{mainnet, florencenet, granadanet}`.
```
Comment (pasqu4le) [inline docs/systemd.md:44]:
```suggestion
For this you'll need a `.service` file to define each systemd service.
The easiest way to get one is to run [`gen_systemd_service_file.py`](../gen_systemd_service_file.py).
You should specify the service name as an argument. Note that there are three
predefined services for `tezos-node`: `tezos-node-{mainnet, florencenet, granadanet}`.
```

File: docs/systemd.md [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+
+E.g.:
+```
+./gen_systemd_service_file.py tezos-node-mainnet
+# or
+./gen_systemd_service_file.py tezos-baker-010-PtGRANAD
+```
+After that you'll have `.service` file in the current directory.
```
Comment (pasqu4le) [inline docs/systemd.md:52]:
```suggestion
After that you'll have a `.service` file in the current directory.
```

File: docs/systemd.md [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+
+Apart from `.service` file you'll need service startup script and default configuration
+file, they can be found in [`scripts`](../docker/package/scripts) and
+[`defaults`](../docker/package/defaults) folders respectively.
```
Comment (pasqu4le) [inline docs/systemd.md:56]:
```suggestion
Apart from the `.service` file you'll need the service startup script and default
configuration file, they can be found in the [`scripts`](../docker/package/scripts) and
[`defaults`](../docker/package/defaults) folders respectively.
```

File: docs/systemd.md [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+
+
+## Multiple similar systemd services
+
+It's possible to run multiple same services, e.g. two `tezos-node`s that run different
```
Comment (pasqu4le) [inline docs/systemd.md:61]:
```suggestion
It's possible to run multiple similar services, e.g. two `tezos-node`s that run different
```

File: README.md (hunk 11 above)
Comment (rvem) [reply README.md:63]: uh, I somehow lost the link :confused:

**Review by pasqu4le — APPROVED** [review]
